### PR TITLE
1110 when no filters provided, report card should notify about it

### LINKED ISSUE #1110: Add a message informing about no active filters or don't return anything from the `format_data_extract`
@kpagacz 

If we have filters we see this:
image

If we don't we see this:
image

Should we update the formatting function in teal.slice to say No filters or remove IRIS etc? And we can't do append_text(cat(datasets$get_formatted_filter_state()) as that outputs to console and we get an app crash as we have a NULL in append_text.

Feel free to spin out to another issue or ignore

_Originally posted by @nikolas-burkoff in https://github.com/insightsengineering/teal/issues/635#issuecomment-1130053893_

### PR DESCRIPTION
**Pull request by m7pr**
Close #1110 

Tested on `teal.modules.general::tm_a_pca()` 

<details><summary> tm_a_pca code </summary>

```r
library(teal.widgets)

data <- teal_data()
data <- within(data, {
  library(nestcolor)
  USArrests <- USArrests
})
datanames(data) <- "USArrests"

app <- init(
  data = data,
  modules = modules(
    tm_a_pca(
      "PCA",
      dat = data_extract_spec(
        dataname = "USArrests",
        select = select_spec(
          choices = variable_choices(
            data = data[["USArrests"]], c("Murder", "Assault", "UrbanPop", "Rape")
          ),
          selected = c("Murder", "Assault"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      ggplot2_args = ggplot2_args(
        labs = list(subtitle = "Plot generated by PCA Module")
      )
    )
  )
)

if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

</details>

Now, when a report card is created, but no filters were specified, we will have an extra text on a report card:

image

A potential change to this solution is to hide the `Filter State` header (and it's empty content) at all.

### CODE REVIEW

File: R/teal_reporter.R [hunk 1]
```diff
@@ -41,7 +41,11 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
     append_fs = function(fs) {
       checkmate::assert_class(fs, "teal_slices")
       self$append_text("Filter State", "header3")
-      self$append_content(TealSlicesBlock$new(fs))
+      if (length(fs)) {
+        self$append_content(TealSlicesBlock$new(fs))
+      } else {
+        self$append_text("No filters specified.")
+      }
```
[the remark this answers is not in the document]
Comment (gogonzo) [reply R/teal_reporter.R:48]: I personally prefer to not show anything - as suggested in the comment

[the remark this answers is not in the document]
Comment (gogonzo) [reply R/teal_reporter.R:48]: But it is more a question to @donyunardi and @lcd2yyz - do you guys prefer to show
`No filters specified` in the outputted document or not showing anything?

[the remark this answers is not in the document]
Comment (lcd2yyz) [reply R/teal_reporter.R:48]: My preference is to keep as-is - explicitly stating no data filter applied.

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**